internal/pkg/config/config.go

```diff
@@ -80,23 +83,28 @@ func InitConfig() {
 	viper.SetConfigType(configFileExtension)
 	viper.AddConfigPath(configFolderPath)
 
-	err = createFolderIfNotExists(configFolderPath)
-	cobra.CheckErr(err)
-	err = createFileIfNotExists(configFilePath)
-	cobra.CheckErr(err)
+	// Write config dir path to global variable
+	FolderPath = configFolderPath
 
 	err = viper.ReadInConfig()
-	cobra.CheckErr(err)
+	if !errors.As(err, &viper.ConfigFileNotFoundError{}) {
+		cobra.CheckErr(err)
+	}
+
+	// This hack is required to allow creating the config file with `viper.WriteConfig`
+	// see https://github.com/spf13/viper/issues/851#issuecomment-789393451
+	viper.SetConfigFile(configFilePath)
+
 	setConfigDefaults()
 
 	viper.AutomaticEnv()
 	viper.SetEnvPrefix("stackit")
 }
 
-func createFolderIfNotExists(folderPath string) error {
-	_, err := os.Stat(folderPath)
+func CreateFolderIfNotExists() error {
```

Comment:
Also this method I think it doesn't need to be public:

```suggestion
func createFolderIfNotExists() error {
```